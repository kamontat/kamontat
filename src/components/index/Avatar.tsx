import React from "react"
import { GatsbyImage, IGatsbyImageData } from "gatsby-plugin-image"
import tw, { styled } from "twin.macro"

import { BaseOptions } from "../../layout/Base"

interface AvatarOptions extends BaseOptions {
  alt?: string
  image?: IGatsbyImageData
}

const Image = styled(GatsbyImage)(() => [tw`w-64 h-auto mx-auto`, tw`rounded md:rounded-lg`])

export const Avatar = ({ image, alt }: AvatarOptions): JSX.Element => {
  if (image) return <Image image={image} alt={alt ?? "unknown"} />
  else return <div />
}
