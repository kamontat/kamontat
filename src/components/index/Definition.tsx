import React from "react"
import tw, { styled } from "twin.macro"

import { BaseOptions } from "../../layout/Base"

interface DefinitionOptions extends BaseOptions {
  message?: string
}

const Message = styled.h2(() => [tw`pb-3 font-sans text-lg`])

export const Definition = ({ message }: DefinitionOptions): JSX.Element => {
  if (message) return <Message>{message}</Message>
  else return <span></span>
}
