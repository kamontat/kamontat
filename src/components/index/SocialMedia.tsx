import React from "react"
import tw from "twin.macro"

import { BaseOptions } from "../../layout/Base"

import { Facebook } from "./socials/Facebook"
import { Github } from "./socials/Github"
import { Instagram } from "./socials/Instagram"
import { Linkedin } from "./socials/Linkedin"
import { Twitter } from "./socials/Twitter"
import { Youtube } from "./socials/Youtube"

export interface SocialObject {
  key: string
  name: string
  url: string

  username: string | undefined
}

export interface SocialOptions extends BaseOptions {
  social: SocialObject[]
}

const SocialBar = tw.div`inline-flex`

export const SocialMedia = ({ social }: SocialOptions): JSX.Element => {
  return (
    <SocialBar>
      {social.reduce((result, s) => {
        const key = s.key.toLowerCase()
        switch (key) {
          case "facebook":
            result.push(<Facebook key={s.key} type={s.key} name={s.name} url={s.url} title={s.username} />)
            break
          case "twitter":
            result.push(<Twitter key={s.key} type={s.key} name={s.name} url={s.url} title={s.username} />)
            break
          case "youtube":
            result.push(<Youtube key={s.key} type={s.key} name={s.name} url={s.url} title={s.username} />)
            break
          case "linkedin":
            result.push(<Linkedin key={s.key} type={s.key} name={s.name} url={s.url} title={s.username} />)
            break
          case "instagram":
            result.push(<Instagram key={s.key} type={s.key} name={s.name} url={s.url} title={s.username} />)
            break
          case "github":
            result.push(<Github key={s.key} type={s.key} name={s.name} url={s.url} title={s.username} />)
            break
        }

        return result
      }, [] as JSX.Element[])}
    </SocialBar>
  )
}
