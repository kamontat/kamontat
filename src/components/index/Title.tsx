import React from "react"
import tw, { styled } from "twin.macro"

import { BaseOptions } from "../../layout/Base"

interface TitleOptions extends BaseOptions {
  name?: string
  shortname?: string
}

// const Name = styled.h1(() => [tw`font-sans text-xl lg:text-2xl `])
const Name = styled.h1(() => [tw`pt-4 m-1`, tw`font-serif text-xl lg:text-2xl`])
const Highlight = styled.span(() => [tw`font-semibold hover:font-bold`])

export const Title = ({ name, shortname }: TitleOptions): JSX.Element => {
  if (name && shortname)
    return (
      <Name>
        <Highlight>{name}</Highlight> ({shortname})
      </Name>
    )
  else if (name) {
    return (
      <Name>
        <Highlight>{name}</Highlight>
      </Name>
    )
  } else if (shortname) {
    return <Name>{shortname}</Name>
  } else {
    return <span></span>
  }
}
