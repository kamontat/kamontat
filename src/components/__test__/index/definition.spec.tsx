import React from "react"
import renderer from "react-test-renderer"
import { Definition } from "../../index/Definition"

describe("ComingSoon", () => {
  it("renders correctly when no input", () => {
    const tree = renderer.create(<Definition />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("renders correctly when message is exist", () => {
    const tree = renderer.create(<Definition message={"Test"} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
