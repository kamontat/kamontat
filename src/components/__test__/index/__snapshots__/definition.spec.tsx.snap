// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComingSoon renders correctly when message is exist 1`] = `
.emotion-0 {
  padding-bottom: 0.75rem;
  font-family: Graphik,sans-serif;
  font-size: 1.125rem;
}

<h2
  className="emotion-0 emotion-1"
>
  Test
</h2>
`;

exports[`ComingSoon renders correctly when no input 1`] = `<span />`;
